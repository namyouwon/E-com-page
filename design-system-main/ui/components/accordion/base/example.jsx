// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import uniqueId from 'lodash.uniqueid';
import Accordion, { AccordionSection } from '../';
import { ActionOverflow } from '../../../components/menus/action-overflow/example';
import Card from '../../cards/';

export const AccordionExample = () => (
  <Accordion>
    <AccordionSection
      summary="Accordion summary"
      isOpen
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - A
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - B
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - C
    </AccordionSection>
  </Accordion>
);

export default <AccordionExample />;

export let states = [
  {
    id: 'section-one-open',
    label: 'Section one open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-two-open',
    label: 'Section two open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-three-open',
    label: 'Section three open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'multiple-sections-open',
    label: 'Multiple sections open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];

export let examples = [
  {
    id: 'long-heading-truncated',
    label: 'Long Section Heading with Truncation for larger screens',
    demoStyles: 'max-width: 500px;',
    storybookStyles: true,
    element: (
      <Card>
        <Accordion>
          <AccordionSection
            summary="Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </Card>
    )
  },
  {
    id: 'nested-accordions',
    label: 'Nested Accordions',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          <Accordion>
            <AccordionSection
              summary="Nested accordion summary"
              isOpen
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - A
            </AccordionSection>
            <AccordionSection
              summary="Nested accordion summary"
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - B
            </AccordionSection>
          </Accordion>
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'styled',
    label: 'Wrapped in Card',
    element: (
      <div className="slds-card">
        <Accordion>
          <AccordionSection
            summary="Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </div>
    )
  },
  {
    id: 'has-action-overflow',
    label: 'Has Action Overflow Menu',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Background Color',
    demoStyles: '--slds-c-accordion-summary-color-background: lightblue',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Text Color',
    demoStyles:
      '--slds-c-accordion-heading-text-color: green; --slds-c-accordion-heading-text-color-hover: red',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];
