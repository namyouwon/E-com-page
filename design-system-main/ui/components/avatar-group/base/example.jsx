// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import {
  AvatarGroup,
  AvatarGroupContent,
  AvatarGrouped,
  AvatarImage,
  AvatarGroupedIcon,
  AvatarGroupedUserInitials
} from '../';

export const ContextInversed = props => (
  <div
    className="demo-only"
    style={{ padding: '0.5rem', background: '#16325c' }}
  >
    {props.children}
  </div>
);

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  }
];

export const examples = [
  {
    id: 'with-icons',
    label: 'With icons',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials',
    label: 'With initials',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials-inversed',
    label: 'With initials inversed',
    demoStyles: 'padding: 0.5rem; background: #16325c',
    storybookStyles: true,
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="user-initials" isInverse />
      </AvatarGroup>
    )
  },
  {
    id: 'with-image-and-icon',
    label: 'With image and icon',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarImage />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarGroupedIcon />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'with-icon-and-user-initials',
    label: 'With icon and user initials',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarGroupedIcon />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarGroupedUserInitials />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials-and-image',
    label: 'With user initials and image',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarGroupedUserInitials />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarImage />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-image',
    label: 'X-Small',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-image',
    label: 'Small',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-image',
    label: 'Medium',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-image',
    label: 'Large',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-icon',
    label: 'X-Small With icon',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-icon',
    label: 'Small With icon',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-icon',
    label: 'Medium With icon',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-icon',
    label: 'Large With icon',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-user-initials',
    label: 'X-Small With initials',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-user-initials',
    label: 'Small With initials',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-user-initials',
    label: 'Medium With initials',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-user-initials',
    label: 'Large With initials',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  }
];
