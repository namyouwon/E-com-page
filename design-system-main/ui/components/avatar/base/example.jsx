// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar, AvatarIcon, AvatarInitials } from '../';

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export { Avatar };

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Avatar>
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  }
];

export let examples = [
  {
    id: 'circle',
    label: 'Circle',
    element: (
      <Avatar className="slds-avatar_circle">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'profile-icon',
    label: 'Profile Avatar Image',
    element: (
      <Avatar className="slds-avatar_profile-image-large">
        <span className="slds-assistive-text">Person name</span>
      </Avatar>
    )
  },
  {
    id: 'group-icon',
    label: 'Group Avatar Image',
    element: (
      <Avatar className="slds-avatar_group-image-large">
        <span className="slds-assistive-text">Group name</span>
      </Avatar>
    )
  },
  {
    id: 'fallback-user-initials',
    label: 'Fallback - User initials',
    element: (
      <Avatar className="slds-avatar_circle">
        <AvatarInitials
          iconClassName="slds-icon-standard-user"
          title="person name"
          initials="AB"
        />
      </Avatar>
    )
  },
  {
    id: 'fallback-user-icon',
    label: 'Fallback - User icon',
    element: (
      <Avatar className="slds-avatar_circle">
        <AvatarIcon symbol="user" />
      </Avatar>
    )
  },
  {
    id: 'fallback-entity-initials',
    label: 'Fallback - Entity Initials',
    element: (
      <Avatar>
        <AvatarInitials
          iconClassName="slds-icon-standard-account"
          title="Company name"
          initials="Ac"
        />
      </Avatar>
    )
  },
  {
    id: 'fallback-entity-icon',
    label: 'Fallback - Entity Icon',
    element: (
      <Avatar>
        <AvatarIcon />
      </Avatar>
    )
  },
  {
    id: 'fallback-user-initials-inverse',
    label: 'Fallback - User Initials Inversed',
    element: (
      <Avatar className="slds-avatar_circle">
        <AvatarInitials title="Person name" initials="AB" isInverse />
      </Avatar>
    )
  },
  {
    id: 'x-small-image',
    label: 'X-Small Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_x-small">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'small-image',
    label: 'Small Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_small">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'medium-image',
    label: 'Medium Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_medium">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },
  {
    id: 'large-image',
    label: 'Large Image',
    element: (
      <Avatar className="slds-avatar_circle slds-avatar_large">
        <img
          alt="Person name"
          src="/assets/images/avatar2.jpg"
          title="Person name"
        />
      </Avatar>
    )
  },

  {
    id: 'linked-avatar',
    label: 'Wrapped in a link',
    element: (
      <a href="#" onClick={e => e.preventDefault()}>
        <Avatar>
          <img
            alt="Person name"
            src="/assets/images/avatar2.jpg"
            title="User avatar"
          />
        </Avatar>
      </a>
    )
  }
];
