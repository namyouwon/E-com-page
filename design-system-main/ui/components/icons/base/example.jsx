// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import PropTypes from 'prop-types';
import SvgIcon from '../../../shared/svg-icon';
import classNames from 'classnames';

export let UtilityIcon = props => {
  const {
    assistiveText,
    className,
    containerClassName,
    isDisabled,
    size,
    title,
    useCurrentColor
  } = props;
  const symbol = props.symbol || 'announcement';
  return (
    <span
      className={classNames(
        'slds-icon_container slds-icon-utility-' + symbol,
        {
          'slds-current-color': useCurrentColor,
          'slds-icon_disabled': isDisabled
        },
        containerClassName
      )}
      title={
        title !== false
          ? title || 'Description of icon when needed'
          : null
      }
    >
      <SvgIcon
        className={classNames(
          'slds-icon',
          {
            'slds-icon_xx-small': size === 'xx-small',
            'slds-icon_x-small': size === 'x-small',
            'slds-icon_small': size === 'small',
            'slds-icon_large': size === 'large'
          },
          className
        )}
        sprite="utility"
        symbol={symbol}
      />
      {assistiveText !== false ? (
        <span className="slds-assistive-text">
          {assistiveText || 'Description of icon when needed'}
        </span>
      ) : null}
    </span>
  );
};
UtilityIcon.displayName = 'UtilityIcon';
UtilityIcon.propTypes = {
  assistiveText: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  /** Controls the size of the icon */
  size: PropTypes.oneOf(['large', 'small', 'x-small', 'xx-small']),
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  /** Sets the slds-current-color class on the container */
  useCurrentColor: PropTypes.bool
};

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Base - Default',
    element: <UtilityIcon className="slds-icon-text-default" />
  }
]

export let examples = [
  {
    id: 'currentColor',
    label: 'Base - Current Color',
    element: (
      <span style={{ color: 'purple' }}>
        <UtilityIcon useCurrentColor />
      </span >
    )
  },
  {
    id: 'rtl-flipped',
    label: 'Base - Horizontally flipped icon (RTL)',
    element: (
      <span dir="rtl">
        <UtilityIcon
          containerClassName="slds-icon_flip"
          className="slds-icon-text-default"
          symbol="picklist_type"
          title="List icon"
          assistiveText="List icon"
        />
      </span>
    )
  },
  {
    id: 'size-xxsmall',
    label: 'Base - size xxsmall',
    element: <UtilityIcon useCurrentColor className="slds-icon_xx-small" />
  },
  {
    id: 'size-xsmall',
    label: 'Base - size xsmall',
    element: <UtilityIcon useCurrentColor className="slds-icon_x-small" />
  },
  {
    id: 'size-small',
    label: 'Base - size small',
    element: <UtilityIcon useCurrentColor className="slds-icon_small" />
  },
  {
    id: 'size-large',
    label: 'Base - size large',
    element: <UtilityIcon useCurrentColor className="slds-icon_large" />
  },
  {
    id: 'disabled',
    label: 'Base - disabled',
    element: <UtilityIcon isDisabled />
  },
  {
    id: 'light',
    label: 'Base - light',
    element: <UtilityIcon className="slds-icon-text-light" />
  },
  {
    id: 'text-success',
    label: 'Base - text success',
    element: <UtilityIcon className="slds-icon-text-success" />
  },
  {
    id: 'text-warning',
    label: 'Base - text warning',
    element: <UtilityIcon className="slds-icon-text-warning" />
  },
  {
    id: 'text-error',
    label: 'Base - text error',
    element: <UtilityIcon className="slds-icon-text-error" />
  }
];
