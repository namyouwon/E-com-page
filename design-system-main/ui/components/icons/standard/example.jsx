// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import PropTypes from 'prop-types';
import SvgIcon from '../../../shared/svg-icon';
import classNames from 'classnames';

export let StandardIcon = props => {
  const {
    assistiveText,
    className,
    containerClassName,
    isDisabled,
    isGrouped,
    title
  } = props;
  const symbol = props.symbol || 'account';
  return (
    <span
      className={classNames(
        `slds-icon_container slds-icon-standard-${symbol}`,
        containerClassName,
        {
          'slds-avatar-grouped__icon': isGrouped,
          'slds-icon_disabled': isDisabled
        }
      )}
      title={
        title !== false ? title || 'Description of icon when needed' : null
      }
    >
      <SvgIcon
        className={classNames('slds-icon', className)}
        sprite="standard"
        symbol={symbol}
      />
      {assistiveText !== false ? (
        <span className="slds-assistive-text">
          {assistiveText || 'Description of icon when needed'}
        </span>
      ) : null}
    </span>
  );
};

StandardIcon.propTypes = {
  isGrouped: PropTypes.bool
};

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default [
  {
    id: 'standard-default',
    label: 'Standard - Default',
    element: <StandardIcon />
  }
]

export let examples = [
  {
    id: 'disabled',
    label: 'Standard - Disabled',
    element: <StandardIcon isDisabled />
  }
]
