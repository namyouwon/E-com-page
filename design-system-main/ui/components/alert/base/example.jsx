// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Alert from '../';
import { UtilityIcon } from '../../icons/base/example';
import EmptyLink from '../../../shared/empty-link/';
import Heading from '../../../shared/heading/';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Alert>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];

export let states = [
  {
    id: 'theme-info',
    label: 'Default (Theme Class  - Deprecated)',
    element: (
      <Alert isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <Alert type="warning">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-warning',
    label: 'Warning (Theme Class  - Deprecated)',
    element: (
      <Alert type="warning" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <Alert type="error">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-error',
    label: 'Error (Theme Class  - Deprecated)',
    element: (
      <Alert type="error" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'offline',
    label: 'Offline',
    element: (
      <Alert type="offline">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-offline',
    label: 'Offline (Theme Class  - Deprecated)',
    element: (
      <Alert type="offline" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];
