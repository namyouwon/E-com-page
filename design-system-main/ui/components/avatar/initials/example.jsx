// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Avatar>
        <AvatarInitials
          iconClassName="slds-icon-standard-account"
          title="company name"
          initials="Ac"
        />
      </Avatar>
    )
  }
];

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="#" onClick={e => e.preventDefault()}>
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
