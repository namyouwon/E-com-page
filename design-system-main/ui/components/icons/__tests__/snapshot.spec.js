// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { ActionIcon } from '../action/example';
import { UtilityIcon } from '../base/example';
const { matchesMarkup } = createHelpers(__dirname);

describe('Icon Action', () => {
  it('renders an x-small action icon', () =>
    matchesMarkup(<ActionIcon className="slds-icon_x-small" />));
  it('renders an xx-small action icon', () =>
    matchesMarkup(<ActionIcon className="slds-icon_xx-small" />));
});

describe('Utility Icon', () => {
  it('should render a utility icon', () => {
    matchesMarkup(<UtilityIcon />);
  });

  it('should render a utility xx-small icon', () => {
    matchesMarkup(<UtilityIcon size="xx-small" />);
  });

  it('should render a utility x-small icon', () => {
    matchesMarkup(<UtilityIcon size="x-small" />);
  });

  it('should render a utility small icon', () => {
    matchesMarkup(<UtilityIcon size="small" />);
  });

  it('should render a utility large icon', () => {
    matchesMarkup(<UtilityIcon size="large" />);
  });

  it('should render a utility icon with currentColor', () => {
    matchesMarkup(<UtilityIcon useCurrentColor />);
  });
});

describe('RTL icons', () => {
  it('should render a horizontally flipped icon', () =>
    matchesMarkup(
      <div dir="rtl" style={{ width: '32px' }}>
        <UtilityIcon
          containerClassName="slds-icon_flip"
          className="slds-icon-text-default"
          symbol="picklist_type"
          title="List icon"
          assistiveText="List icon"
        />
      </div>
    ));
});
