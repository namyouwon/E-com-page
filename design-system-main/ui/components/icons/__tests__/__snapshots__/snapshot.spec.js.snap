// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Action renders an x-small action icon 1`] = `
<span
  className="slds-icon_container slds-icon_container_circle slds-icon-action-description"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#description"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Icon Action renders an xx-small action icon 1`] = `
<span
  className="slds-icon_container slds-icon_container_circle slds-icon-action-description"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small"
  >
    <use
      xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#description"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`RTL icons should render a horizontally flipped icon 1`] = `
<div
  dir="rtl"
  style={
    Object {
      "width": "32px",
    }
  }
>
  <span
    className="slds-icon_container slds-icon-utility-picklist_type slds-icon_flip"
    title="List icon"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#picklist_type"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      List icon
    </span>
  </span>
</div>
`;

exports[`Utility Icon should render a utility icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility icon with currentColor 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement slds-current-color"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility large icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_large"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility small icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility x-small icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;

exports[`Utility Icon should render a utility xx-small icon 1`] = `
<span
  className="slds-icon_container slds-icon-utility-announcement"
  title="Description of icon when needed"
>
  <svg
    aria-hidden={true}
    className="slds-icon slds-icon_xx-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#announcement"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon when needed
  </span>
</span>
`;
